File: lib/models/tipo_evento.dart
class TipoEvento {
  final int id;
  final String codigo;
  final String nombre;
  final String? descripcion;
  final String? icono;
  final String? color;
  final int orden;
  final bool activo;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  TipoEvento({
    required this.id,
    required this.codigo,
    required this.nombre,
    this.descripcion,
    this.icono,
    this.color,
    required this.orden,
    required this.activo,
    this.createdAt,
    this.updatedAt,
  });

  factory TipoEvento.fromJson(Map<String, dynamic> json) {
    return TipoEvento(
      id: json['id'] as int,
      codigo: json['codigo'] as String,
      nombre: json['nombre'] as String,
      descripcion: json['descripcion'] as String?,
      icono: json['icono'] as String?,
      color: json['color'] as String?,
      orden: json['orden'] as int? ?? 0,
      activo: json['activo'] == 1 || json['activo'] == true,
      createdAt:
          json['created_at'] != null
              ? DateTime.parse(json['created_at'] as String)
              : null,
      updatedAt:
          json['updated_at'] != null
              ? DateTime.parse(json['updated_at'] as String)
              : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'codigo': codigo,
      'nombre': nombre,
      'descripcion': descripcion,
      'icono': icono,
      'color': color,
      'orden': orden,
      'activo': activo,
      'created_at': createdAt?.toIso8601String(),
      'updated_at': updatedAt?.toIso8601String(),
    };
  }
}

File: app/Models/TipoEvento.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class TipoEvento extends Model
{
    use HasFactory, SoftDeletes;

    protected $table = 'tipos_evento';

    protected $fillable = [
        'codigo',
        'nombre',
        'descripcion',
        'icono',
        'color',
        'orden',
        'activo',
    ];

    protected $casts = [
        'activo' => 'boolean',
        'orden' => 'integer',
    ];

    /**
     * Relación con eventos
     */
    public function eventos()
    {
        return $this->hasMany(Evento::class, 'tipo_evento_id');
    }

    /**
     * Scope para tipos activos
     */
    public function scopeActivos($query)
    {
        return $query->where('activo', true);
    }

    /**
     * Scope para ordenar por orden
     */
    public function scopeOrdenados($query)
    {
        return $query->orderBy('orden')->orderBy('nombre');
    }
}
